pkg/resource/db_cluster/custom_update.go

```diff
@@ -561,7 +562,12 @@ func (rm *resourceManager) newCustomUpdateRequestPayload(
 		res.SetEnableIAMDatabaseAuthentication(*desired.ko.Spec.EnableIAMDatabaseAuthentication)
 	}
 	if desired.ko.Spec.EngineVersion != nil && delta.DifferentAt("Spec.EngineVersion") {
```

Comment:
Can we move this logic into a little function called `requireEngineVersionUpdate` or something similar? that was we could add unit tests for it :)